### PR TITLE
Fix DQN target update interval for multi-env

### LINKED ISSUE #1373: [Bug]: loading DQN changes target_update_interval
### 🐛 Bug

when loading a DQN model, target_update_interval gets divided by n_envs. So saving and loading a DQN model changes target_update_interval.

in dqn, line 160:
self.target_update_interval = max(self.target_update_interval // self.n_envs, 1)

### To Reproduce

```
import gym
from stable_baselines3 import DQN
from stable_baselines3.common.vec_env import DummyVecEnv

env = DummyVecEnv([lambda: gym.make("CartPole-v1") for _ in range(4)])
model = DQN("MlpPolicy", env, verbose=1)

for _ in range(5):
    print(model.target_update_interval)
    model.save("dqn_cartpole")
    del model
    model = DQN.load("dqn_cartpole")
```

### Relevant log output / Error message

```shell
2500
625
156
39
9
```

### System Info

_No response_

### Checklist

- [X] I have checked that there is no similar [issue](https://github.com/DLR-RM/stable-baselines3/issues) in the repo
- [X] I have read the [documentation](https://stable-baselines3.readthedocs.io/en/master/)
- [X] I have provided a minimal working example to reproduce the bug
- [X] I've used the [markdown code blocks](https://help.github.com/en/articles/creating-and-highlighting-code-blocks) for both code and stack traces.

### PR DESCRIPTION
**Pull request by tobirohrer**
## Description

closes #1373 (Target interval for DQN changes when saving and reloading the model)

The root cause lies in the function `_setup_model()` in `dqn/dqn.py`.
When the model is set up, the target update interval is updated by:
`self.target_update_interval = max(self.target_update_interval // self.n_envs, 1)`

This causes the `target_update_interval` to minimize by a factor of `n_envs` whenever the model is saved and reloaded.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)

## Checklist

- [x] I've read the [CONTRIBUTION](https://github.com/DLR-RM/stable-baselines3/blob/master/CONTRIBUTING.md) guide (**required**)
- [x] I have updated the changelog accordingly (**required**).
- [ ] My change requires a change to the documentation.
- [x] I have updated the tests accordingly (*required for a bug fix or a new feature*).
- [ ] I have updated the documentation accordingly.
- [ ] I have opened an associated PR on the [SB3-Contrib repository](https://github.com/Stable-Baselines-Team/stable-baselines3-contrib) (if necessary)
- [ ] I have opened an associated PR on the [RL-Zoo3 repository](https://github.com/DLR-RM/rl-baselines3-zoo) (if necessary)
- [x] I have reformatted the code using `make format` (**required**)
- [x] I have checked the codestyle using `make check-codestyle` and `make lint` (**required**)
- [x] I have ensured `make pytest` and `make type` both pass. (**required**)
- [x] I have checked that the documentation builds using `make doc` (**required**)

Note: You can run most of the checks using `make commit-checks`.

Note: we are using a maximum length of 127 characters per line

### CODE REVIEW

**Comment by Pythoniasm** [comment]
This is a common issue across Stable Baselines3 regarding multi-envs. Same issue on checkpoint saves and across algorithms (not just DQN from my experience).

**Comment by araffin** [comment]
> How things are calculated and executed regarding multi-envs is inconsistent in general and is a common issue across Stable Baselines3

That's true, main reason being that multi envs support happened later and also same granularity of control is not possible with multi env (for instance, for the callbacks, every call to a callback correspond to `n_envs` steps).

That's why there is the RL Zoo which tries to make everything consistent being the scene.

File: tests/test_save_load.py [hunk 1]
```diff
@@ -730,3 +730,13 @@ def test_load_invalid_object(tmp_path):
     with warnings.catch_warnings(record=True) as record:
         PPO.load(path, custom_objects=dict(learning_rate=lambda _: 1.0))
     assert len(record) == 0
+
+
+def test_dqn_target_update_interval():
```
Comment (araffin) [inline tests/test_save_load.py:735]: please use a tmp path (see other examples)

File: tests/test_save_load.py [hunk 2]
```diff
@@ -730,3 +730,13 @@ def test_load_invalid_object(tmp_path):
...
+    # `target_update_interval` should not change when reloading the model. See GH Issue #1373.
+    env = DummyVecEnv([lambda: gym.make("CartPole-v1") for _ in range(2)])
```
Comment (araffin) [inline tests/test_save_load.py:737]: `make_vec_env` should be prefered here

File: stable_baselines3/dqn/dqn.py [hunk 3]
```diff
@@ -174,7 +172,7 @@ def _on_step(self) -> None:
         This method is called in ``collect_rollouts()`` after each step in the environment.
         """
         self._n_calls += 1
-        if self._n_calls % self.target_update_interval == 0:
+        if self._n_calls % max(self.target_update_interval // self.n_envs, 1) == 0:
```
Comment (araffin) [inline stable_baselines3/dqn/dqn.py:175]: you should move the comment
```python
        # Account for multiple environments
        # each call to step() corresponds to n_envs transitions
```

here

**Review by araffin — APPROVED** [review]
LGTM, thanks =)
Could you do a similar PR to SB3 contrib for QR-DQN?

**Review by qgallouedec — APPROVED** [review]

**Comment by araffin** [comment]
>  Could you do a similar PR to SB3 contrib for QR-DQN?

@tobirohrer in case you didn't see

**Pull request merged**